### PR TITLE
Libdawn shared

### PR DESCRIPTION
**Pull request by Kangz**
PTAL this makes libdawn (dawn.c and dawncpp.cpp) a shared library.

This is the first of several PRs that will make the CMakeLists build close to what we'll want the GN build to be.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,6 +156,12 @@ function(DawnInternalTarget folder target)
     set_property(TARGET ${target} APPEND PROPERTY COMPILE_OPTIONS ${DAWN_INTERNAL_FLAGS})
     set_property(TARGET ${target} APPEND PROPERTY COMPILE_DEFINITIONS ${DAWN_INTERNAL_DEFS})
 
+    # Common include directories shared by all internal targets
+    target_include_directories(${target} PRIVATE ${SRC_DIR} ${GENERATED_DIR} ${INCLUDE_DIR})
+
+    # All internal targets requires the headers to have been generated
```
Comment (SenorBlanco) [inline CMakeLists.txt]: Nit: s/requires/require/ ?

Comment (Kangz) [reply CMakeLists.txt]: Done (DAWN_DAWN)

Comment (SenorBlanco) [reply CMakeLists.txt]: https://media.giphy.com/media/kKdgdeuO2M08M/giphy.gif

**Comment by Kangz** [comment]
Thanks for the reviews!

**Pull request merged**